### DISCUSSION TITLE
Why do I get a (1 != 1) appended to my SQL query debug log

### DISCUSSION THREAD

**Discussion by ManiMozaffar** [sqlalchemy/sqlalchemy#12173] [Usage Questions]
Hey,
I have this piece of code that i'm trying to run
<img width="563" alt="Screenshot 2024-12-12 at 14 29 46" src="https://github.com/user-attachments/assets/4fa6268b-2edf-4270-8acc-257f803a306b" />

From logs I see sometimes things like this:
<img width="620" alt="Screenshot 2024-12-12 at 14 30 41" src="https://github.com/user-attachments/assets/3cf41e74-00fa-4ef6-b0f7-f21b3911b6c3" />

But the query should have been translated to what log message is showing except the "(1 != 1)" statement which is clearly not in my code. I tried looking into github and document and couldn't find anything for this behavior, is this something in my end or is this an optimization or some other thing I don't understand? 
I find it dangerous if that's what actually executed on database, given that 1 != 1 is False therefore my where condition becomes False.

**Comment by zzzeek** [comment]
it means `wb_ids` is an empty collection, see https://docs.sqlalchemy.org/en/20/core/metadata.html#sqlalchemy.schema.Column.in_

**Reply by ManiMozaffar** [reply]
Thanks for quick answer! :)
